Week_3/homework/index.html

```diff
@@ -0,0 +1,296 @@
+						// after recieving the converted JSON data, pass the JSON to the renderSearchResults() function
+						renderSearchResults(data)
+					});
+			}
+		}
+
+		// function that renders the search results as a unordered list
+		function renderSearchResults(searchResults) {
+				// selects the unordered list element search-results-list
+				const ul = document.querySelector('#search-results-list')
+				// shows the unordered list if was hidden previously
+				ul.classList.remove("hidden");
+				// clears out any list items from the previous search
+				ul.innerHTML = ''
+				// loops through each search result and creates and attaches a list item for the unordered list
+				searchResults.forEach((searchResult, index) => {
+					// creates a new unordered list element
+					const li = document.createElement('li')
+					// sets the list item's class as search-result
+					li.setAttribute('class', 'search-result')
+					// sets the text inside the list item as the name and state of the city 
+					const fullName = searchResult.name + ', ' + searchResult.state
+					li.innerHTML = fullName
+					// if the list item of a city is clicked, call the selectCity() function
+					li.addEventListener('click', () => selectCity(fullName, searchResult.name, searchResult.state, searchResult.lat, searchResult.lon))
+					// attaches the list item elements to search-results-list
+					ul.appendChild(li)
+			})	
+		}
+
+		// function that is called whenever a city has been selected
+		function selectCity(fullName, name, state, lat, lon) {
+			// hides the search-results-list since it is not needed right now
+			document.querySelector('#search-results-list').className = 'hidden'
+			// sets the global city variable
+			document.querySelector("#search-input").value = ''
+
+			city = {
+				fullName: fullName,
+				name: name,
+				state: state,
+				lat: lat,
+				lon: lon
+			}
+            // BEGIN CODING HERE
+			getData(city);
+			
+		}
+
+		function renderHeading(city, date, temp, weather, icon) {
+			document.querySelector('#location').innerText = 'Weather for ' + city.fullName;
+			document.querySelector('#today').innerText = date;
+			document.querySelector('#temp').innerText = temp + '°';
+			document.querySelector('#weather').innerText = weather;
+			document.querySelector('#image').src = 'icons/' + icon + '.svg';
+		}
+
+
+		function getData(city) {
+			let apiCall = `https://api.openweathermap.org/data/2.5/onecall?lat=${city.lat}&lon=${city.lon}&units=imperial&appid=${apiKey}`;
+			// calls the API
+				fetch(apiCall)
+					.then((response) => 
+						response.json()
+					)
+					.then((data) => {
+						weather = (data.current.weather)[0].main;
+						temp = data.current.temp;
+						day = new Date(data.daily[0].dt * 1000);
+						date = weekday[day.getDay()] + ' ' + day.getDate();
+						icon = data.current.weather[0].icon;
+						renderHeading(city, date, temp, weather, icon);
+						
+
+					    day1_lowtemp = (data.daily[1].temp.min).toFixed(0);

```

Comment:
These lines can definitely be updated with a for loop with something like the following

for(let i=0; i<5; i++){
  dayi_lowtemp = data.daily[i].temp.min.toFixed(0);
  dayi_hightemp = data.daily[i].temp.max.toFixed(0);
  dayi = new Date(data.daily[i].dt * 1000);
  datei = weekday[dayi.getDay()] + ' ' + dayi.getDate();
  imgi = data.daily[i].weather[0].icon;
  renderData(datei, imgi, dayi_lowtemp, dayi_hightemp, i);
}